### PR TITLE
Add `Users` and `Group` related functions

### PR DESCRIPTION
**Pull request by otavio**
This was a collaborative work between Johannes Schilling
<dario@deaktualisierung.org>, Fredrick Brennan <copypaste@kittens.ph>
and myself.

This PR is a split-off from #864 so we merge ready parts first. Next, we work on the iterators.

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

### CODE REVIEW

**Comment by asomers** [comment]
So what's the plan?  What do you expect to add in a follow up PR?

**Comment by otavio** [comment]
@asomers we did not include the iterators here and I'd like to extend it after this is merged as it is independent and allow for reducing the code to work/review.

Do you mind to go over this here so we clean any leftover and merge this one?

**Comment by otavio** [comment]
@asomers any comment on this one?

File: src/unistd.rs [hunk 1]
```diff
@@ -2424,3 +2424,263 @@ pub fn access<P: ?Sized + NixPath>(path: &P, amode: AccessFlags) -> Result<()> {
...
+            Ok(None) | Err(_) => buflimit as usize,
+        };
+
+        let mut cbuf = Vec::with_capacity(bufsize);
+        let mut pwd = mem::MaybeUninit::<libc::passwd>::uninit();
+        let mut res = ptr::null_mut();
+
+        loop {
+            let error = unsafe {
+                Errno::clear();
```
Comment (asomers) [inline src/unistd.rs]: Is this really necessary?

File: src/unistd.rs [hunk 2]
```diff
@@ -2424,3 +2424,263 @@ pub fn access<P: ?Sized + NixPath>(path: &P, amode: AccessFlags) -> Result<()> {
...
+            Ok(None) | Err(_) => buflimit as usize,
+        };
+
+        let mut cbuf = Vec::with_capacity(bufsize);
+        let mut grp = mem::MaybeUninit::<libc::group>::uninit();
+        let mut res = ptr::null_mut();
+
+        loop {
+            let error = unsafe {
+                Errno::clear();
```
Comment (asomers) [inline src/unistd.rs]: Is `Errno::clear` really necessary?

File: src/unistd.rs (hunk 1 above)
Comment (otavio) [reply src/unistd.rs]: Not really.

File: src/unistd.rs (hunk 2 above)
Comment (otavio) [reply src/unistd.rs]: Not really.

**Comment by otavio** [comment]
@asomers I made the changes. Please let me know if something else is still needed.

**Comment by otavio** [comment]
> Now that you no longer `Errno::clear`, you need to remove some `unsafe` blocks, apparently.

Just did; sorry for missing that.

**Comment by otavio** [comment]
@asomers fixed the unnecessary `unsafe` block. Should be good to go.

**Pull request merged**